### PR TITLE
PATCH /session/account

### LINKED ISSUE #53: Implement PATCH /session/account
The account update route is currently not implemented at all:
http://docs.accountjsonapi.apiary.io/#reference/current-user/account/update

---

- [x] claim the issue (@ianstalter123)
- [x] Start PR (#63)
- [ ] Add tests to [tests/integration/routes/account-test.js](https://github.com/hoodiehq/hoodie-server-account/blob/master/tests/integration/routes/account-test.js)
- [ ] Implement the new route in [routes/acount.js](https://github.com/hoodiehq/hoodie-server-account/blob/master/routes/account.js) (not accounts.js, that’s for admins only)
- [ ] review & merge
- [ ] update https://github.com/gr2m/milestones/issues/88
- [ ] [uncomment integration test](https://github.com/hoodiehq/hoodie-app-something-tracker/blob/6c8b0f5a8aecb8fd3645bbb23782f6d3effd66af/test/account-test.js#L112-L136)

### PR DESCRIPTION
**Pull request by gr2m**
follow up for #54, closes #53

### CODE REVIEW

**Comment by ianstalter123** [comment]
@gr2m cool I will check through it thx !!- - -

File: routes/account.js [hunk 1]
```diff
@@ -102,6 +102,50 @@ function accountRoutes (server, options, next) {
...
+      return sessions.find(sessionId)
+
+      .then(function (session) {
+        return accounts.update(session.account, {
+          username: newUsername,
+          password: newPassword
+        }, {
+          include: request.query.include
+        })
+      })
```
Comment (gr2m) [inline routes/account.js]: we need to rearrange the code right now. Throwing an error on line 120 will not stop the code flow, only an "Unhandled error" would be logged to the console, the `return sessions.find(sessionId)` would be executed anyway. 

You can do it like in the `GET /session/account` handler:

```diff
- admins.validateSession(sessionId)
- .then(function (doc) {
-   throw errors.FORBIDDEN_ADMIN_ACCOUNT
- })
- var newUsername = request.payload.data.attributes.username
- var newPassword = request.payload.data.attributes.password
- 
- return sessions.find(sessionId)

+ var newUsername = request.payload.data.attributes.username
+ var newPassword = request.payload.data.attributes.password
+ admins.validateSession(sessionId)
+ 
+ .then(function (doc) {
+   throw errors.FORBIDDEN_ADMIN_ACCOUNT
+ })
+ 
+ .catch(function (error) {
+   if (error.name === 'not_found') {
+     return sessions.find(sessionId)
+   }
+ 
+   throw error
+ })
```

That way `session.find(sessionId)` only gets executed if session is not by an admin. Makes sense?

**Comment by gr2m** [comment]
@ianstalter123 is this good for review, or are you still working on it?

For trailing whitespace, most editors have a configuration where you can ask it to remove trailing slashes. Most editors also support [editorconfig](http://editorconfig.org/). I have one in my home directory so it always applies as default, it looks like this:

```
$ cat ~/.editorconfig 
root = true

[*]
end_of_line = lf
insert_final_newline = true
indent_size = 2
trim_trailing_whitespace = true
[*.md]
trim_trailing_whitespace = false
insert_final_newline = false
[*.txt]
trim_trailing_whitespace = false
insert_final_newline = false
[*.hbs]
trim_trailing_whitespace = false
insert_final_newline = false
[*.json]
trim_trailing_whitespace = false
insert_final_newline = false
indent_size = 4
[package.json]
indent_size = 2
```

**Comment by ianstalter123** [comment]
@gr2m I installed editor config, not sure how it works (is there a shortcut or does it automatically just set) -> if you can fix this and close please do otherwise I will try to get rid of trailing line spaces again tommorow

**Comment by gr2m** [comment]
You need to download a plugin for your editor here:
http://editorconfig.org/#download

I’ve update your PR, I only did minor adjustments for standard and a little simplification for the double route mocking. Thanks, great work @ianstalter123 :+1:

**Pull request merged**